src/System.ObjectModel/tests/ReadOnlyDictionary/ReadOnlyDictionaryTests.cs

```diff
@@ -0,0 +1,715 @@
+                KeyValuePair<TKey, TValue> tempItem;
+
+                // Verify we have not gotten more items then we expected                
+                Assert.True(iterations < expectedCount,
+                    "Err_9844awpa More items have been returned fromt the enumerator(" + iterations + " items) then are in the expectedElements(" + expectedCount + " items)");
+
+                // Verify Current returned the correct value
+                itemFound = false;
+
+                for (int i = 0; i < itemsVisited.Length; ++i)
+                {
+                    if (!itemsVisited[i] && currentItem.Equals(expectedItems[i]))
+                    {
+                        itemsVisited[i] = true;
+                        itemFound = true;
+                        break;
+                    }
+                }
+                Assert.True(itemFound, "Err_1432pauy Current returned unexpected value=" + currentItem);
+
+                // Verify Current always returns the same value every time it is called
+                for (int i = 0; i < 3; i++)
+                {
+                    tempItem = enumerator.Current;
+                    Assert.Equal(currentItem, tempItem);
+                }
+
+                iterations++;
+            }
+
+            for (int i = 0; i < expectedCount; ++i)
+            {
+                Assert.True(itemsVisited[i], "Err_052848ahiedoi Expected Current to return true for item: " + expectedItems[i] + "index: " + i);
+            }
+
+            Assert.Equal(expectedCount, iterations);
+
+            for (int i = 0; i < 3; i++)
+            {
+                Assert.False(enumerator.MoveNext(), "Err_2929ahiea Expected MoveNext to return false after" + iterations + " iterations");
+            }
+
+            // Verify non deterministic behavior of current every time it is called after the enumerator is positioned after the last item
+            for (int i = 0; i < 3; i++)
+            {
+                try
+                {
+                    KeyValuePair<TKey, TValue> tempCurrent = enumerator.Current;
+                }
+                catch (Exception) { }
+            }
+
+            enumerator.Dispose();
+        }
+
+        /// <summary>
+        /// Verifies that the non-generic enumerator retrieves the correct items.
+        /// </summary>
+        private void VerifyEnumerator(IReadOnlyDictionary<TKey, TValue> collection, KeyValuePair<TKey, TValue>[] expectedItems)
+        {
+            IEnumerator enumerator = collection.GetEnumerator();
+            int iterations = 0;
+            int expectedCount = expectedItems.Length;
+
+            // Verify non deterministic behavior of current every time it is called before a call to MoveNext() has been made			
+            for (int i = 0; i < 3; i++)
+            {
+                try
+                {
+                    object tempCurrent = enumerator.Current;
+                }
+                catch (Exception) { }
```

Comment:
This bit of code is strange because it currently isn't validating anything and `enumerator.Current` at this point is expected to be Undefined.  I am not sure what I can do here to validate Undefined and I am inclined to just remove this loop as it doesn't appear to be adding any value to the test.

Thoughts?